src/get_v2_vs_frac.py

```diff
@@ -0,0 +1,298 @@
+        SetObjectStyle(gFracVsV2[-1], linecolor=kAzure+4, linewidth=2, markerstyle=kFullCircle, markersize=1, markercolor=kAzure+4)
+
+        print(f"Processing pt bin {iPt+1}/{nPtBins}: {ptMin:.2f} < pT < {ptMax:.2f} GeV/c, nSets: {nSets}")
+        avrV2XErrL.append(Double_t(sum(gV2[i].GetErrorXlow(iPt) for i in range(nSets)) / nSets))
+        avrV2XErrH.append(Double_t(sum(gV2[i].GetErrorXhigh(iPt) for i in range(nSets)) / nSets))
+        
+
+        v2Values = [hV2[i].GetBinContent(iPt + 1) for i in range(nSets)]
+        v2Unc = [hV2[i].GetBinError(iPt + 1) for i in range(nSets)]
+        fracFDValues = [hFracFD[i].GetBinContent(iPt + 1) for i in range(nSets)]
+        fracFDUnc = [hFracFD[i].GetBinError(iPt + 1) for i in range(nSets)]
+
+        for iSet, (v2, fracFD, v2Unc, fracFDUnc) in enumerate(zip(v2Values, fracFDValues, v2Unc, fracFDUnc)):
+            print(f"pt: {ptCent:.4f}, v2: {v2:.4f}, fracFD: {fracFD:.4f}")
+            gFracVsV2[iPt].SetPoint(iSet, fracFD, v2)
+            gFracVsV2[iPt].SetPointError(iSet, fracFDUnc, v2Unc)
+        
+        # gFracVsV2Fit = TGraphErrors(gFracVsV2[-1])
+        linFunc = TF1("linear", "pol1", 0, 1)
+        SetObjectStyle(linFunc, color=kOrange+1, linestyle=9, linewidth=2)
+        gFracVsV2[-1].Fit("linear", "", "", 0, 1)
+        chi2 = linFunc.GetChisquare()
+        ndf = linFunc.GetNDF()
+
+        # get the confidence intervals 0.683
+        fitter = TVirtualFitter.GetFitter()
+        fitter.GetConfidenceIntervals(hV2VsFrac[-1], 0.683)
+        hV2VsFrac[-1].SetLineColorAlpha(kAzure+5, 0.15)
+
+        # get the v2 value at the FD fraction = 1, and it is not the last bin?
+        hV2VsPtFD.SetBinContent(iPt + 1, 
+                                hV2VsFrac[-1].GetBinContent(hV2VsFrac[-1].GetNbinsX()))
+        hV2VsPtFD.SetBinError(iPt + 1,
+                                hV2VsFrac[-1].GetBinError(hV2VsFrac[-1].GetNbinsX()))
+        
+        # get the v2 value at the FD fraction = 0, and it is the first bin?
+        hV2VsPtPrompt.SetBinContent(iPt + 1, 
+                                    hV2VsFrac[-1].GetBinContent(hV2VsFrac[-1].GetBin(1)))
+        hV2VsPtPrompt.SetBinError(iPt + 1,
+                                    hV2VsFrac[-1].GetBinError(hV2VsFrac[-1].GetBin(1)))
+        
+        #TODO: plot the v2 vs pt, and the center of the pt bin is calculate by the average of pT
+
+        ptStrings.append(f"{ptMin:.1f} < #it{{p}}_{{T}} < {ptMax:.1f} GeV/#it{{c}}")
+        chi2Strings.append(f"#chi^{{2}}/n.d.f = {chi2:.2f}/{ndf:.2f}")
+
+
+    # save the results
+    parentDir = os.path.dirname(fracFiles[0])
+    dirName = os.path.basename(parentDir).replace('frac', 'v2VsFrac')
+    fileName = os.path.basename(fracFiles[0]).replace('frac', 'v2VsFrac')
+    outFileName = os.path.join(parentDir, dirName, fileName)
+    outputdir = os.path.dirname(outFileName)
+    os.makedirs(outputdir, exist_ok=True)
+    outFile = TFile(outFileName, "recreate")
```

Comment:
Can't this just be replaced by
```suggestion
    outputdir = os.path.dirname(fracFiles[0]).replace('frac', 'v2VsFrac')
    os.makedirs(outputdir, exist_ok=True)
    outFile = TFile(f"{outputdir}/v2VsFrac.root", "recreate")
```

Reply:
Here, I would like to avoid replacing it directly. I think replacing 'frac' or other 'key' words directly is not safe. What if they exist in the path. But you are right, some cases we can just directly use the name, since there is no `suffix` anymore.